specs/aws-c-io.spec

```diff
@@ -0,0 +1,77 @@
+Name:           aws-c-io
+Version:        0.10.12 
+Release:        1%{?dist}
+Summary:        IO package for AWS SDK for C
+Epoch:          1
+
+License:        ASL-2.0
+URL:            https://github.com/awslabs/%{name}
+Source0:        %{url}/archive/v%{version}/%{name}-%{version}.tar.gz
+
+BuildRequires:  gcc
+BuildRequires:  cmake
+BuildRequires:  s2n-tls-devel
+BuildRequires:  aws-c-common-devel = 1:0.6.14
+BuildRequires:  aws-c-cal-devel = 1:0.5.12
+
+Requires:       s2n-tls-libs
+Requires:       aws-c-common-libs = 1:0.6.14
+Requires:       aws-c-cal-libs = 1:0.5.12
+
+%description
+IO package for AWS SDK for C. It handles all IO and TLS work
+for application protocols.
+
+
+%package libs
+Summary:        IO package for AWS SDK for C
+Requires:       %{name}%{?_isa} = %{epoch}:%{version}-%{release}
+	
+%description libs
+IO package for AWS SDK for C. It handles all IO and TLS work
+for application protocols.
+
+
+%package devel
+Summary:        IO package for AWS SDK for C
+Requires:       %{name}-libs%{?_isa} = %{epoch}:%{version}-%{release}
+Requires:       s2n-tls-devel
+Requires:       aws-c-cal-devel = 1:0.5.12
```

Comment:
There are a couple of times where I declared the `devel` build requires as the requires for its `devel` package in order to build everything correctly.

For example, if one of the packages required `aws-c-io-devel` but did not directly declare a build dependency on `s2n-tls-devel` or `aws-c-cal-devel`, it would fail to build saying that it could not locate `s2n` or `aws-c-cal` even though none of the source for that package would reference those libraries directly. It seemed no matter what it required the files that came from the `devel` packages for those sub-dependencies.

So I was not sure if it was better either:
1. Declare the `devel` dependencies as `Requires` to the `devel` package as done here. So that packages that consume this `devel` package do not have to worry about having to explicitly include them.
2. Explicitly declare the `devel` sub-dependencies even if they are not directly used in source. Like I did for `aws-c-cal-devel` and `openssl-devel` in `aws-c-auth`: https://github.com/davdunc/awscli-2-rpm/pull/12/files#diff-20e8789e449584712e53ac562af0a929f32c64d13e748d62755828d72a9181b4R13-R16. 

Feedback here would be great.